### ISSUE TITLE
Action timing out as of a few days ago

### ISSUE THREAD

**Issue by elie222** [cypress-io/github-action#235]
I see some others having similar issues as of a few days ago but despite having upgraded to v2.3.9 of this plugin we're still seeing a timeout:

```
Starting the development server...

http://localhost:3000 timed out
Error: Timeout awaiting 'request' for 1000ms
```

Any ideas?

Full logs:

```
Run cypress-io/github-action@v2.3.9
Received 83886080 of 173170025 (48.4%), 80.0 MBs/sec
Received 146800640 of 173170025 (84.8%), 70.0 MBs/sec
Received 168975721 of 173170025 (97.6%), 53.7 MBs/sec
Received 168975721 of 173170025 (97.6%), 40.3 MBs/sec
Received 168975721 of 173170025 (97.6%), 32.2 MBs/sec
Received 168975721 of 173170025 (97.6%), 26.8 MBs/sec
Received 168975721 of 173170025 (97.6%), 23.0 MBs/sec
Received 168975721 of 173170025 (97.6%), 20.1 MBs/sec
Received 173170025 of 173170025 (100.0%), 18.1 MBs/sec
Cache Size: ~165 MB (173170025 B)
/bin/tar --use-compress-program zstd -d -xf /home/runner/work/_temp/395f82d7-f931-4439-a531-26fcc49358d3/cache.tzst -P -C /home/runner/work/frontwork/frontwork
/usr/bin/yarn --frozen-lockfile
yarn install v1.22.5
[1/4] Resolving packages...
[2/4] Fetching packages...
info fsevents@2.2.1: The platform "linux" is incompatible with this module.
info "fsevents@2.2.1" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@1.2.13: The platform "linux" is incompatible with this module.
info "fsevents@1.2.13" is an optional dependency and failed compatibility check. Excluding it from installation.
info fsevents@2.1.3: The platform "linux" is incompatible with this module.
info "fsevents@2.1.3" is an optional dependency and failed compatibility check. Excluding it from installation.
[3/4] Linking dependencies...
warning " > @testing-library/user-event@12.2.2" has unmet peer dependency "@testing-library/dom@>=7.21.4".
warning " > react-image-annotation@0.9.10" has unmet peer dependency "prop-types@^15.6.0".
warning "@storybook/react > @babel/preset-react@7.12.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@storybook/addon-essentials > @storybook/addon-docs > @babel/plugin-transform-react-jsx@7.12.5" has unmet peer dependency "@babel/core@^7.0.0-0".
warning " > styled-components@5.2.1" has unmet peer dependency "react-is@>= 16.8.0".
warning " > @graphql-codegen/typescript-react-apollo@2.0.7" has unmet peer dependency "graphql-tag@^2.0.0".
warning " > @storybook/addon-essentials@6.0.28" has unmet peer dependency "babel-loader@^8.0.0".
warning " > @storybook/addon-essentials@6.0.28" has unmet peer dependency "react-is@^16.8.0".
warning "@storybook/addon-essentials > @storybook/addon-docs@6.0.28" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@storybook/addon-essentials > @storybook/addon-docs@6.0.28" has unmet peer dependency "babel-loader@^8.0.0".
warning "@storybook/addon-essentials > @storybook/addon-docs@6.0.28" has unmet peer dependency "react-is@^16.8.0".
warning "@storybook/react > @storybook/core@6.0.28" has unmet peer dependency "@babel/core@*".
warning "@storybook/addon-essentials > @storybook/addon-docs > @storybook/core > @babel/plugin-proposal-export-default-from@7.12.1" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@storybook/addon-essentials > @storybook/addon-docs > @storybook/core > @babel/register@7.12.1" has unmet peer dependency "@babel/core@^7.0.0-0".
warning "@storybook/addon-essentials > @storybook/addon-docs > @storybook/core > @babel/plugin-proposal-export-default-from > @babel/plugin-syntax-export-default-from@7.12.1" has unmet peer dependency "@babel/core@^7.0.0-0".
warning " > @storybook/preset-create-react-app@3.1.5" has unmet peer dependency "@babel/core@*".
warning " > @storybook/preset-create-react-app@3.1.5" has unmet peer dependency "@storybook/node-logger@*".
warning " > @storybook/react@6.0.28" has unmet peer dependency "@babel/core@^7.0.1".
warning "@storybook/react > @babel/preset-flow@7.12.1" has unmet peer dependency "@babel/core@^7.0.0-0".
warning " > storybook-react-router@1.0.8" has unmet peer dependency "@storybook/addon-actions@^4.0.0||^5.0.0||5.2.0-beta.13".
warning " > storybook-react-router@1.0.8" has unmet peer dependency "react-router@^4.0.0||^5.0.0".
[4/4] Building fresh packages...
Done in 46.55s.
/opt/hostedtoolcache/node/12.19.0/x64/bin/npx cypress cache list
┌─────────┬──────────────┐
│ version │ last used    │
├─────────┼──────────────┤
│ 5.6.0   │ a minute ago │
└─────────┴──────────────┘
/opt/hostedtoolcache/node/12.19.0/x64/bin/npx cypress verify

25l[19:17:37]  Verifying Cypress can run /home/runner/.cache/Cypress/5.6.0/Cypress [started]
[19:17:40]  Verifying Cypress can run /home/runner/.cache/Cypress/5.6.0/Cypress [completed]
25h25h::debug::saving NPM modules
/bin/tar --posix --use-compress-program zstd -T0 -cf cache.tzst -P -C /home/runner/work/frontwork/frontwork --files-from manifest.txt
Cache saved successfully
/bin/tar --posix --use-compress-program zstd -T0 -cf cache.tzst -P -C /home/runner/work/frontwork/frontwork --files-from manifest.txt
Cache saved successfully
start server with command "yarn start"
current working directory "/home/runner/work/frontwork/frontwork/client"
waiting on "http://localhost:3000" with timeout of 30 seconds
/usr/bin/yarn start
yarn run v1.22.5
$ npm run watch:css && cross-env EXTEND_ESLINT=true react-app-rewired start
npm WARN lifecycle The node binary used for scripts is /tmp/yarn--1605467898479-0.33537419860463835/node but npm is using /opt/hostedtoolcache/node/12.19.0/x64/bin/node itself. Use the `--scripts-prepend-node-path` option to include the path for the node binary npm was executed with.

> frontwork@0.1.0 watch:css /home/runner/work/frontwork/frontwork/client
> postcss src/tailwind.css -o src/main.css

ℹ ｢wds｣: Project is running at http://10.1.0.4
ℹ ｢wds｣: webpack output is served from 
ℹ ｢wds｣: Content not from webpack is served from /home/runner/work/frontwork/frontwork/client/public
ℹ ｢wds｣: 404s will fallback to /
Starting the development server...

http://localhost:3000 timed out
Error: Timeout awaiting 'request' for 1000ms
```

**Comment by benjick** [comment]
`v2.3.9` seems to work for me (commented out my whole job except checkout and cypress-action though). `v2.3.10` gave me the same error. Just pushed the whole workflow with `v2.3.9` now

**Comment by elie222** [comment]
My config. (Was working till a few days ago with `cypress-io/github-action@v2`):

```
      - name: Cypress run
        uses: cypress-io/github-action@v2.3.9
        with:
          record: true
          working-directory: ./client
          start: yarn start
          wait-on: 'http://localhost:3000'
          wait-on-timeout: 30
        env:
          # pass the Dashboard record key as an environment variable
          CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
          # pass GitHub token to allow accurately detecting a build vs a re-run build
          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by thien-duplexpoint** [comment]
```
Run cypress-io/github-action@v2
Received 77358328 of 77358328 (100.0%), 98.6 MBs/sec
Cache Size: ~74 MB (77358328 B)
/bin/tar --use-compress-program zstd -d -xf /home/runner/work/_temp/6fa8658f-55f9-4a4e-b973-f14b2ea1638e/cache.tzst -P -C /home/runner/work/partnership/partnership
Received 174686969 of 174686969 (100.0%), 151.2 MBs/sec
Cache Size: ~167 MB (174686969 B)
/bin/tar --use-compress-program zstd -d -xf /home/runner/work/_temp/703e5dc5-97ea-4be6-bdc2-b2b3d1a44e1c/cache.tzst -P -C /home/runner/work/partnership/partnership
/usr/bin/yarn --frozen-lockfile
yarn install v1.22.5
[1/4] Resolving packages...
success Already up-to-date.
Done in 0.56s.
/opt/hostedtoolcache/node/10.13.0/x64/bin/npx cypress cache list
┌─────────┬───────────────────┐
│ version │ last used         │
├─────────┼───────────────────┤
│ 5.4.0   │ a few seconds ago │
└─────────┴───────────────────┘
start server with command "bundle exec rails s -p 3000 -b 0.0.0.0"
current working directory "/home/runner/work/partnership/partnership"
waiting on "http://localhost:3000" with timeout of 60 seconds
/opt/hostedtoolcache/Ruby/2.6.5/x64/bin/bundle exec rails s -p 3000 -b 0.0.0.0
=> Booting Puma
=> Rails 6.0.3.3 application starting in test 
=> Run `rails server --help` for more startup options
WARNING: Stimulus Reflex requires caching to be enabled. Caching allows the session to be modified during ActionCable requests.
To enable caching in development, run:
    rails dev:cache
[4269] Puma starting in cluster mode...
[4269] * Version 4.3.5 (ruby 2.6.5-p114), codename: Mysterious Traveller
[4269] * Min threads: 5, max threads: 5
[4269] * Environment: test
[4269] * Process workers: 2
[4269] * Preloading application
[4269] * Listening on tcp://0.0.0.0:3000
[4269] Use Ctrl-C to stop
[4269] - Worker 0 (pid: 4278) booted, phase: 0
[4269] - Worker 1 (pid: 4281) booted, phase: 0
DEPRECATION WARNING: autoprefixer-rails was deprected. Migration guide:
https://github.com/ai/autoprefixer-rails/wiki/Deprecated (called from process at /home/runner/work/partnership/partnership/vendor/bundle/ruby/2.6.0/gems/autoprefixer-rails-9.8.6.3/lib/autoprefixer-rails/processor.rb:36)
DEPRECATION WARNING: autoprefixer-rails was deprected. Migration guide:
https://github.com/ai/autoprefixer-rails/wiki/Deprecated (called from process at /home/runner/work/partnership/partnership/vendor/bundle/ruby/2.6.0/gems/autoprefixer-rails-9.8.6.3/lib/autoprefixer-rails/processor.rb:36)
http://localhost:3000 timed out
Error: Timeout awaiting 'request' for 1000ms
```
I got the same error.
I think the new version is the cause of this break. It just happened last weekend.

**Comment by benjick** [comment]
I used this as a workaround:
```
- name: Wait for server to start up
  run: timeout 240 bash -c 'while [[ "$(curl -s -o /dev/null -w ''%{http_code}'' localhost:9090)" != "200" ]]; do sleep 5; done' || false

- name: Run e2e tests
  uses: cypress-io/github-action@v2.3.9
  with:
    config-file: dev/tests/cypress.json
    record: true
  env:
    CYPRESS_RECORD_KEY: ${{ secrets.CYPRESS_RECORD_KEY }}
    GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
```

**Comment by bahmutov** [comment]
Please provide either a reproducible example or run with GH debug logs enabled https://github.com/cypress-io/github-action#debugging to see what the code is doing

**Comment by daiky00** [comment]
@bahmutov I am having the same issue

```

Run cypress-io/github-action@v2
Received 106124923 of 106124923 (100.0%), 103.0 MBs/sec
Cache Size: ~101 MB (106124923 B)
Received 125829120 of 173719723 (72.4%), 119.9 MBs/sec
/bin/tar --use-compress-program zstd -d -xf /home/runner/work/_temp/69b98ccf-f0a5-43ac-84cb-349ae9c667e9/cache.tzst -P -C /home/runner/work/frontend/frontend
Received 173719723 of 173719723 (100.0%), 130.5 MBs/sec
Cache Size: ~166 MB (173719723 B)
/bin/tar --use-compress-program zstd -d -xf /home/runner/work/_temp/6406d584-0c85-4cbd-b5de-c54ee75e71d7/cache.tzst -P -C /home/runner/work/frontend/frontend
/usr/bin/yarn --frozen-lockfile
yarn install v1.22.5
[1/4] Resolving packages...
success Already up-to-date.
Done in 1.27s.
/opt/hostedtoolcache/node/12.13.0/x64/bin/npx cypress cache list
┌─────────┬───────────────────┐
│ version │ last used         │
├─────────┼───────────────────┤
│ 5.2.0   │ a few seconds ago │
└─────────┴───────────────────┘
start server with command "yarn run staging"
current working directory "/home/runner/work/frontend/frontend"
waiting on "http://localhost:3000" with timeout of 60 seconds
/usr/bin/yarn run staging
yarn run v1.22.5
$ REACT_APP_LOCAL_STAGING=true react-scripts -r @cypress/instrument-cra start
ℹ ｢wds｣: Project is running at http://10.1.0.4/
ℹ ｢wds｣: webpack output is served from 
ℹ ｢wds｣: Content not from webpack is served from /home/runner/work/frontend/frontend/public
ℹ ｢wds｣: 404s will fallback to /
Starting the development server...

http://localhost:3000 timed out
Error: Timeout awaiting 'request' for 1000ms
```

**Comment by dfalling** [comment]
I'm having the same problem as @daiky00 and @elie222: when Cypress launches my app, it's not listening on port 3000 as expected, but rather at address `http://10.1.0.4`.  I had this problem previously because of [react-scripts@3.4.1](https://github.com/facebook/create-react-app/issues/8688).

This time it showed up as soon as I switched to `v2` of this action due to Github's removal of `v1`'s functionality.

**Comment by bahmutov** [comment]
Without a reproducible example, there is nothing we can do to debug this. I have added Webpack-dev-server-example in https://github.com/cypress-io/github-action/blob/master/.github/workflows/example-webpack.yml and it seems to run just fine. Closing as non-reproducible, but will be happy to re-open if a reproducible example is provided.

**Comment by elie222** [comment]
So myself, benjick, thien-duplexpoint, dfalling, daiky00 have all faced this issue and the thread is only 2 days old but it's already being closed because a reproduction wasn't provided within 24 hours?
Seems like it's far too early to close this issue which is clearly a real issue.

**Comment by yoonwaiyan** [comment]
I'm having this issue as well, and I agree it's too early to close this issue.

**Comment by jennifer-shehane** [comment]
We still read issues if they are closed - and are happy to reopen if someone comments with a reproducible example *at any time*. This is just our process in handling issues. It does not mean the issue is being ignored.

**Comment by barisbora** [comment]
When I switch back to @v2.3.8 problem gone!

**Comment by bahmutov** [comment]
I would be happy to reopen the issue if someone provides a reproducible example and/or debug logs https://github.com/cypress-io/github-action#debugging

**Comment by yoonwaiyan** [comment]
Ok. Here's the debug log to compare between working and non-working versions:

<details>
<summary>v2.3.8 (working)</summary>

```shell
##[debug]parsed command: bundle exec rails server -e test -p 5017
##[debug]found command "/opt/hostedtoolcache/Ruby/2.6.6/x64/bin/bundle"
##[debug]with arguments exec rails server -e test -p 5017
##[debug]running "/opt/hostedtoolcache/Ruby/2.6.6/x64/bin/bundle" exec rails server -e test -p 5017 in /home/runner/work/my/project/spec
##[debug]without waiting for the promise to resolve
waiting on "http://localhost:5017" with timeout of 120 seconds
/opt/hostedtoolcache/Ruby/2.6.6/x64/bin/bundle exec rails server -e test -p 5017
##[debug]18ms GET localhost:5017 ECONNREFUSED
##[debug]1022ms GET localhost:5017 ECONNREFUSED
##[debug]2027ms GET localhost:5017 ECONNREFUSED
##[debug]3029ms GET localhost:5017 ECONNREFUSED
##[debug]4032ms GET localhost:5017 ECONNREFUSED
=> Booting Puma
=> Rails 5.2.4.3 application starting in test 
=> Run `rails server -h` for more startup options
##[debug]5034ms GET localhost:5017 ECONNREFUSED
##[debug]6037ms GET localhost:5017 ECONNREFUSED
##[debug]7038ms GET localhost:5017 ECONNREFUSED
##[debug]8040ms GET localhost:5017 ECONNREFUSED
##[debug]9043ms GET localhost:5017 ECONNREFUSED
##[debug]10045ms GET localhost:5017 ECONNREFUSED
##[debug]11048ms GET localhost:5017 ECONNREFUSED
Puma starting in single mode...
* Version 3.12.6 (ruby 2.6.6-p146), codename: Llamas in Pajamas
* Min threads: 5, max threads: 5
* Environment: test
* Listening on tcp://0.0.0.0:5017
Use Ctrl-C to stop
[Webpacker] Browserslist: caniuse-lite is outdated. Please run:
npx browserslist@latest --update-db
```
</details>

And here's my debug log for the newest version, which is not working.
<details>
<summary>v2 (not working)</summary>

```shell
Run cypress-io/github-action@v2
##[debug]working directory /home/runner/work/my/project/spec
##[debug]using custom Cypress cache folder "/home/runner/.cache/Cypress"
Skipping install because install parameter is false
##[debug]Separated start commands bundle exec rails server -e test -p 5017
start server "bundle exec rails server -e test -p 5017 with command "bundle exec rails server -e test -p 5017"
current working directory "/home/runner/work/my/project/spec"
##[debug]parsed command: bundle exec rails server -e test -p 5017
waiting on "http://localhost:5017" with timeout of 120 seconds
##[debug]Waiting for urls http://localhost:5017
##[debug]Waiting for url http://localhost:5017
##[debug]19ms undefined undefined ECONNREFUSED
##[debug]found command "/opt/hostedtoolcache/Ruby/2.6.6/x64/bin/bundle"
##[debug]with arguments exec rails server -e test -p 5017
##[debug]running "/opt/hostedtoolcache/Ruby/2.6.6/x64/bin/bundle" exec rails server -e test -p 5017 in /home/runner/work/my/project/spec
##[debug]without waiting for the promise to resolve
/opt/hostedtoolcache/Ruby/2.6.6/x64/bin/bundle exec rails server -e test -p 5017
##[debug]1022ms undefined undefined ECONNREFUSED
##[debug]2025ms undefined undefined ECONNREFUSED
##[debug]3027ms undefined undefined ECONNREFUSED
##[debug]4029ms undefined undefined ECONNREFUSED
=> Booting Puma
=> Rails 5.2.4.3 application starting in test 
=> Run `rails server -h` for more startup options
##[debug]5031ms undefined undefined ECONNREFUSED
##[debug]6033ms undefined undefined ECONNREFUSED
##[debug]7034ms undefined undefined ECONNREFUSED
##[debug]8037ms undefined undefined ECONNREFUSED
##[debug]9039ms undefined undefined ECONNREFUSED
##[debug]10041ms undefined undefined ECONNREFUSED
Puma starting in single mode...
* Version 3.12.6 (ruby 2.6.6-p146), codename: Llamas in Pajamas
* Min threads: 5, max threads: 5
* Environment: test
* Listening on tcp://0.0.0.0:5017
Use Ctrl-C to stop
##[debug]12549ms undefined undefined ETIMEDOUT
##[debug]14554ms undefined undefined ETIMEDOUT
##[debug]16557ms undefined undefined ETIMEDOUT
##[debug]18560ms undefined undefined ETIMEDOUT
##[debug]20563ms undefined undefined ETIMEDOUT
##[debug]22566ms undefined undefined ETIMEDOUT
##[debug]24568ms undefined undefined ETIMEDOUT
##[debug]26572ms undefined undefined ETIMEDOUT
##[debug]28576ms undefined undefined ETIMEDOUT
##[debug]30580ms undefined undefined ETIMEDOUT
##[debug]32584ms undefined undefined ETIMEDOUT
##[debug]34588ms undefined undefined ETIMEDOUT
##[debug]36590ms undefined undefined ETIMEDOUT
##[debug]38592ms undefined undefined ETIMEDOUT
##[debug]40595ms undefined undefined ETIMEDOUT
##[debug]42599ms undefined undefined ETIMEDOUT
##[debug]44601ms undefined undefined ETIMEDOUT
##[debug]46604ms undefined undefined ETIMEDOUT
##[debug]48608ms undefined undefined ETIMEDOUT
##[debug]50613ms undefined undefined ETIMEDOUT
##[debug]52616ms undefined undefined ETIMEDOUT
##[debug]54619ms undefined undefined ETIMEDOUT
##[debug]56622ms undefined undefined ETIMEDOUT
##[debug]58624ms undefined undefined ETIMEDOUT
##[debug]60627ms undefined undefined ETIMEDOUT
##[debug]62629ms undefined undefined ETIMEDOUT
##[debug]64632ms undefined undefined ETIMEDOUT
##[debug]66636ms undefined undefined ETIMEDOUT
##[debug]68639ms undefined undefined ETIMEDOUT
##[debug]70642ms undefined undefined ETIMEDOUT
##[debug]72648ms undefined undefined ETIMEDOUT
##[debug]74652ms undefined undefined ETIMEDOUT
##[debug]76656ms undefined undefined ETIMEDOUT
##[debug]78661ms undefined undefined ETIMEDOUT
##[debug]80664ms undefined undefined ETIMEDOUT
##[debug]82670ms undefined undefined ETIMEDOUT
##[debug]84672ms undefined undefined ETIMEDOUT
##[debug]86678ms undefined undefined ETIMEDOUT
##[debug]88680ms undefined undefined ETIMEDOUT
##[debug]90683ms undefined undefined ETIMEDOUT
##[debug]92686ms undefined undefined ETIMEDOUT
##[debug]94688ms undefined undefined ETIMEDOUT
##[debug]96691ms undefined undefined ETIMEDOUT
##[debug]98696ms undefined undefined ETIMEDOUT
##[debug]100700ms undefined undefined ETIMEDOUT
##[debug]102703ms undefined undefined ETIMEDOUT
##[debug]104705ms undefined undefined ETIMEDOUT
##[debug]106708ms undefined undefined ETIMEDOUT
##[debug]108711ms undefined undefined ETIMEDOUT
##[debug]110715ms undefined undefined ETIMEDOUT
##[debug]112718ms undefined undefined ETIMEDOUT
##[debug]114722ms undefined undefined ETIMEDOUT
##[debug]116726ms undefined undefined ETIMEDOUT
##[debug]118730ms undefined undefined ETIMEDOUT
http://localhost:5017 timed out
##[debug]120734ms undefined undefined ETIMEDOUT
##[debug]Timeout awaiting 'request' for 1000ms
##[debug]RequestError: Timeout awaiting 'request' for 1000ms
##[debug]    at ClientRequest.<anonymous> (/home/runner/work/_actions/cypress-io/github-action/v2/dist/index.js:3542:65)
##[debug]    at Object.onceWrapper (events.js:300:26)
##[debug]    at ClientRequest.emit (events.js:215:7)
##[debug]    at ClientRequest.origin.emit (/home/runner/work/_actions/cypress-io/github-action/v2/dist/index.js:59917:20)
##[debug]    at Socket.socketErrorListener (_http_client.js:406:9)
##[debug]    at Socket.emit (events.js:210:5)
##[debug]    at emitErrorNT (internal/streams/destroy.js:92:8)
##[debug]    at emitErrorAndCloseNT (internal/streams/destroy.js:60:3)
##[debug]    at processTicksAndRejections (internal/process/task_queues.js:80:21)
##[debug]    at Timeout.timeoutHandler [as _onTimeout] (/home/runner/work/_actions/cypress-io/github-action/v2/dist/index.js:77879:25)
##[debug]    at listOnTimeout (internal/timers.js:533:17)
##[debug]    at processTimers (internal/timers.js:475:7)
Error: Timeout awaiting 'request' for 1000ms
##[debug]Node Action run completed with exit code 1
##[debug]Finishing: Cypress.io
```
</details>

From what I understand from the log, my Rails server has already started after 10 seconds but somehow the newest wait-on implementation is giving ETIMEDOUT and waited for 120 seconds as configured in `wait-on-timeout`.

This is the config: if it's helpful:

```yaml
- name: Cypress.io
  uses: cypress-io/github-action@v2
  with:
    install: false
    start: bundle exec rails server -e test -p 5017
    working-directory: spec
    wait-on: 'http://localhost:5017'
    wait-on-timeout: 120
    browser: chrome
    headless: true
```

**Comment by bahmutov** [comment]
@yoonwaiyan I don't know why the rails server would not respond in this case, would something like this affect it? https://github.com/sharetribe/sharetribe/issues/1725#issuecomment-182691908

Since I don't use Rails, can you create the simplest "Hello World" repo that starts the server and tries to run Cypress (and is not working) for my to clone and play with?

**Comment by elie222** [comment]
We had the issue with Create React App. We've also reverted to 2.3.8 for now which works 👍

**Comment by yoonwaiyan** [comment]
To elaborate more on my logs, my Rails server is working perfectly fine on
both versions, while Cypress tests run on v2.3.8 but not started on newest
version. There’s no code change other than switching between versions -
it’s on the same pull request in my repo. I think this issue is not related
to the framework we use...

**Comment by bahmutov** [comment]
@elie222 and @yoonwaiyan I have added a way to run `src/ping` yourself to investigate what is going on and why the server is not responding https://github.com/cypress-io/github-action#debugging I suggest you give it a try and see if you can find why `got ...` module is not working with your server. I will try adding a react-scripts example to make sure it works

**Comment by bahmutov** [comment]
@elie222 I have added an example with react-scripts@3 here https://github.com/cypress-io/github-action/tree/master/examples/react-scripts which seems to pass the action just fine https://github.com/cypress-io/github-action/actions/runs/372517200 Not sure what is about your project that might be different, maybe some custom settings where the server does not respond? Try debugging it using my comment above with `node src/ping-cli`

**Comment by yoonwaiyan** [comment]
I think this issue is more likely to happen on larger codebases where build time or load time may take significantly longer than new/smaller repos. It could be helpful if there's a way to simulate that but I'm not sure is there any way to do that without manually dumping dependencies.

**Comment by bahmutov** [comment]
It might be, but it is still weird that we don't hit this problem in example repos. One thing that might help is implementing the https://github.com/cypress-io/github-action/issues/243 that I just opened

**Comment by bahmutov** [comment]
Hey everyone, I have added support for custom "waiting" commands in https://github.com/cypress-io/github-action/releases/tag/v2.6.0 Maybe you can use some external package like https://github.com/jeffbski/wait-on to wait for the server to respond with more luck

**Comment by jrparish** [comment]
@bahmutov - It appears the `timeout` property of the `got` request might be causing the issue. If the index page doesn't respond within 1 second, it times out and retries. This probably explains why the test examples work fine but real projects fail. 
Do you remember why this was added? 
Reference - https://github.com/cypress-io/github-action/compare/master...jrparish:master

**Comment by bahmutov** [comment]
That might be it @jrparish - I have added a test with server that takes 5 seconds and it used to fail, now it is passing in https://github.com/cypress-io/github-action/pull/245

**Comment by elie222** [comment]
I don't think it has been resolved. I'm going back to 2.3.8 again 🤷‍♂️

![image](https://user-images.githubusercontent.com/3090527/99905129-84a61600-2cd7-11eb-90de-66f2d0e6dcb5.png)

**Comment by eywolfe** [comment]
I also don't think this has been resolved!  Switching to 2.3.8 doesn't work for me either.  I don't have a reproducible example, but wanted to comment in case others are experiencing this.

**Comment by bahmutov** [comment]
I feel you, I would love to see a reproducible example

**Comment by eywolfe** [comment]
Turns out my issue was unrelated to cypress-io/github-action!  If others are having network timeouts that are not fixed by switching back to v2.3.8, check that you are specifying the node version in your github actions workflow.  If you don't specify the version it uses whatever github considers the default, which changed around the time that v2.3.9 was released 😄 I figured I'd comment here since this issue describes the exact symptoms I saw!
